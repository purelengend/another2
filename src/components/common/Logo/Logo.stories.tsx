import type { Meta, StoryObj } from '@storybook/react'
import { BrowserRouter } from 'react-router-dom'
import Logo from './index'

const meta = {
  title: 'Practice Two/Logo Component',
  component: Logo,
  decorators: [
    Story => (
      <BrowserRouter>
        <div
          style={{
            backgroundColor: '#532EE7',
            padding: '10px',
          }}
        >
          <Story />
        </div>
      </BrowserRouter>
    ),
  ],
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {
    textColor: { control: 'color' },
  },
} satisfies Meta<typeof Logo>

export default meta
type Story = StoryObj<typeof meta>

export const LogoWithText: Story = {
  args: {
    text: 'Agility',
    textSize: 'md',
  },
}

export const LogoWithImage: Story = {
  args: {
    imageSrc: 'src/assets/images/logo_website.webp',
    widthSize: '100px',
    heightSize: '40px',
  },
}
