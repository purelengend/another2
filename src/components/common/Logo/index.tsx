import React from 'react'
import { Flex, Heading, Image } from '@chakra-ui/react'
import { Link } from 'react-router-dom'

interface LogoProps {
  text?: string
  textColor?: string
  textSize?: string
  imageSrc?: string
  widthSize?: string | number
  heightSize?: string | number
}

const Logo: React.FC<LogoProps> = ({
  text,
  textColor = 'black',
  textSize = 'md',
  imageSrc,
  widthSize = '120px',
  heightSize = '40px',
}) => {
  return (
    <Link to="/">
      <Flex align="center" className="logo" cursor="pointer">
        {/* Render the text if provided */}
        {text && (
          <Heading as="h1" size={textSize} color={textColor}>
            {text}
          </Heading>
        )}
        {/* Render the image if provided */}
        {imageSrc && (
          <Image
            className="logo-image"
            loading="eager"
            srcSet={imageSrc}
            src={imageSrc}
            alt="The logo website"
            width={widthSize}
            height={heightSize}
          />
        )}
      </Flex>
    </Link>
  )
}

export default Logo
