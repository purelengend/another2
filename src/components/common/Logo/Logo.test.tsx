import { render } from '@testing-library/react'
import Logo from '.'
import { MemoryRouter } from 'react-router-dom'

describe('Logo component', () => {
  it('renders without crashing', () => {
    const { container } = render(
      <MemoryRouter>
        <Logo />
      </MemoryRouter>
    )
    expect(container).toMatchSnapshot()
  })

  it('renders text when text prop is provided', () => {
    const { container } = render(
      <MemoryRouter>
        <Logo text="My Logo" />
      </MemoryRouter>
    )
    expect(container).toMatchSnapshot()
  })

  it('renders image when imageSrc prop is provided', () => {
    const { container } = render(
      <MemoryRouter>
        <Logo imageSrc="logo.png" />
      </MemoryRouter>
    )
    expect(container).toMatchSnapshot()
  })

  it('applies custom text color and size', () => {
    const { getByText } = render(
      <MemoryRouter>
        <Logo text="My Custom Logo" textColor="red" textSize="lg" />
      </MemoryRouter>
    )
    const logoText = getByText('My Custom Logo')
    expect(logoText).toHaveStyle('color: red')
    expect(logoText).toHaveStyle('font-size: 2em')
  })

  it('applies custom image width and height', () => {
    const { getByAltText } = render(
      <MemoryRouter>
        <Logo imageSrc="logo.png" widthSize="150px" heightSize="50px" />
      </MemoryRouter>
    )
    const logoImage = getByAltText('The logo website')
    expect(logoImage).toHaveStyle('width: 150px')
    expect(logoImage).toHaveStyle('height: 50px')
  })

  it('renders both text and image when both props are provided', () => {
    const { getByText, getByAltText } = render(
      <MemoryRouter>
        <Logo text="My Combo Logo" imageSrc="logo.png" />
      </MemoryRouter>
    )
    expect(getByText('My Combo Logo')).toBeInTheDocument()
    expect(getByAltText('The logo website')).toBeInTheDocument()
  })
})
