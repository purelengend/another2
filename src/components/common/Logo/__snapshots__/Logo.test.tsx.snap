// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component renders image when imageSrc prop is provided 1`] = `
<div>
  <a
    href="/"
  >
    <div
      class="logo css-759u60"
    >
      <img
        alt="The logo website"
        class="logo-image css-1x2q8y0"
        loading="eager"
        src="logo.png"
        srcset="logo.png"
      />
    </div>
  </a>
</div>
`;

exports[`Logo component renders text when text prop is provided 1`] = `
<div>
  <a
    href="/"
  >
    <div
      class="logo css-759u60"
    >
      <h1
        class="chakra-heading css-1xu3tth"
      >
        My Logo
      </h1>
    </div>
  </a>
</div>
`;

exports[`Logo component renders without crashing 1`] = `
<div>
  <a
    href="/"
  >
    <div
      class="logo css-759u60"
    />
  </a>
</div>
`;
